File: mc-test/mc_pi.jl
function estimate_pi(N::Int)
    nc = 0
    for _ in 1:N
        x, y = rand(), rand()
        if hypot(x, y) < 1
            nc += 1
        end
    end
    return nc / N * 4
end

t0 = time()
N = 10000000
pi_estimate = estimate_pi(N)
t1 = time()

println(pi_estimate, "\t", t1 - t0)

File: mc-test/mc_pi.py
from random import random
from time import time

import numpy as np

t0 = time()
N = 10000000
nc = 0
for i in range(N):
    x, y = random(), random()
    if np.hypot(x, y) < 1:
        nc = nc + 1
pi_estimate = nc / N * 4
t1 = time()
t = t1 - t0
print(pi_estimate, "\t", t)
